### ISSUE TITLE
GeneratedPluginRegistrant.java:24: error: package net.jonhanson.flutter_native_splash does not exist

### ISSUE THREAD

**Issue by iwontknow** [jonbhanson/flutter_native_splash#645]
**Describe the bug**

I can successfully create a Splash Screen with the `dart run flutter_native_splash:create` command. However, if i try to run the app afterwards. I get an Error and the build will fail. 

> *CensoredPath*\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java:24: error: package net.jonhanson.flutter_native_splash does not exist flutterEngine.getPlugins().add(new net.jonhanson.flutter_native_splash.FlutterNativeSplashPlugin());

**Configuration**
```yaml
flutter_native_splash:
  color: "#42a5f5"

android_12:
     color: "#42a5f5"
     fullscreen: true
```
**Device (please complete the following information):**
 - Device: Pixel 7 Pro (Emulator)
 - OS: Android API 33

**To Reproduce**
1. Install the dependencies, pub get
2. Fill in the configuration for dependencie
3. Create Splash Screen (dart run flutter_native_splash:create)
4. Build app
5. Get Error

**Screenshots**
-

**Additional context**
I tried following, but nothing helped. I also searched for similar bug related to this dependencie but found nothing. Did i miss something? 
`flutter clean`
`flutter pub cache repair`
Uninstall / Reinstall dependencies
Create Splashscreen, delete Splashscreen
Invalidate Caches from IDE, Repair IDE, Restart IDE

**Comment by jonbhanson** [comment]
Maybe you put the package dependency in `dev_dependencies` but you're using the package methods in your app?  If so, move the dependency to `dependencies` instead.

**Comment by iwontknow** [comment]
Hey, I didn't use the methods in my app yet, but I tried both. First `dev_dependencies` and then `dependencies`.

**Comment by jonbhanson** [comment]
Can you supply a minimal project that demonstrates the error?

**Comment by iwontknow** [comment]
So I have an update for you. After a restart of my computer and `flutter clean && flutter pub get`, it magically works. I didnt change a thing. Sorry for any inconvenience I may have caused you.

**Comment by jonbhanson** [comment]
Glad you were able to resolve it. 👍

**Comment by lukehutch** [comment]
@jonbhanson I have the same problem (`flutter clean && flutter pub get` doesn't fix it for me).

When I add to `pubspec.yaml`

```yaml
dev_dependencies: 
  flutter_native_splash: ^2.3.9
```

then the following lines get added to `GeneratedPluginRegistrant.java`:

```java
    try {
      flutterEngine.getPlugins().add(new net.jonhanson.flutter_native_splash.FlutterNativeSplashPlugin());
    } catch (Exception e) {
      Log.e(TAG, "Error registering plugin flutter_native_splash, net.jonhanson.flutter_native_splash.FlutterNativeSplashPlugin", e);
    }
```

and on build I get the error:

```console
android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java:49: error: package net.jonhanson.flutter_native_splash does not exist
      flutterEngine.getPlugins().add(new net.jonhanson.flutter_native_splash.FlutterNativeSplashPlugin());
```

Doing `grep -r net.jonhanson.flutter_native_splash .` fails to find this package mentioned anywhere but `GeneratedPluginRegistrant`, whereas searching for the package names of other plugins in `GeneratedPluginRegistrant` shows numerous paths.

Since I'm adding this to `dev_dependencies`, I don't think this plugin should even be being added to `GeneratedPluginRegistrant`...

**Comment by nnsyu** [comment]
@iwontknow Oh.. I've had the same thing, and as you say, reboot pc and `flutter clean -> flutter pub get` so it goes well... It's ironic

**Comment by FuadEfendi** [comment]
and it failed after restart & clean & get:

```

* What went wrong:
Execution failed for task ':app:compileReleaseJavaWithJavac'.
> Compilation failed; see the compiler output below.
  warning: [options] source value 8 is obsolete and will be removed in a future release
  warning: [options] target value 8 is obsolete and will be removed in a future release
  warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
  /Users/fe/Projects/flutter/cscexams_v2/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java:94: error: package net.jonhanson.flutter_native_splash does not exist
        flutterEngine.getPlugins().add(new net.jonhanson.flutter_native_splash.FlutterNativeSplashPlugin());
                                                                              ^
  1 error
  3 warnings
```

**Comment by FuadEfendi** [comment]
This is really reproducible. I am typically running script which will bump build number in `pubspec.yaml`:

```bash
#!/bin/bash
sed -i '' -E "s/(version: [0-9.]+\+)[0-9]+/\1$(git rev-list --count HEAD)/" pubspec.yaml
git commit -am "version bump"
flutter build appbundle --obfuscate --split-debug-info=debug/2.0.3
```

So, it fails. Restart dowsn't help. But I narrowed issue: when I run regular build (after bumping version) and only after that obfuscated version, it does not fail:

```bash
#!/bin/bash
sed -i '' -E "s/(version: [0-9.]+\+)[0-9]+/\1$(git rev-list --count HEAD)/" pubspec.yaml
git commit -am "version bump"
flutter build appbundle 
flutter build appbundle --obfuscate --split-debug-info=debug/2.0.3
```

Why is that?

ANother issue I noticed, when I just introduced this integration, when I tried to upgrade application, first run froze with black screen on iOS; subsequent upgrades don't have any issues. It was only when upgrade from `without flutter_native_splash' to `with flutter_native_splash`.

And third issue is that on some Android devices my custom splash is cut-out on top, it is inside circle and top is cut out.

**Comment by andrew2558** [comment]
I encountered the same issue. After restarting, running flutter clean, and flutter pub get, the same error still occurs.

**Comment by iampato** [comment]
Same error on `flutter_native_splash: ^2.4.6`

**Comment by nathenxbrewer** [comment]
Happening to me with 2.4.6. This started happening a few days ago. 

I've removed it from dev_dependencies and I was able to build.
